### ISSUE TITLE
Setting the layout overflow alone does not work

### ISSUE THREAD

**Issue by jasongrout** [jupyter-widgets/ipywidgets#1842]
```python
from ipywidgets import *
VBox([IntSlider(), IntSlider()], layout={'overflow': 'hidden', 'width':'200px', 'height':'50px'})
```
gives

<img width="933" alt="screen shot 2017-11-27 at 5 00 15 pm" src="https://user-images.githubusercontent.com/192614/33297190-92a7c572-d394-11e7-9d9c-373de6dd72c3.png">
(notice the overflow is auto, the default, not 'hidden' like we wanted)

The code at https://github.com/jupyter-widgets/ipywidgets/blob/34091c58f5f21a947503031d48f623a23fa58b86/packages/base/src/widget_layout.ts#L100 means that having both `overflow`, `overflow_x`, and `overflow_y` traits means they stomp on each other, i.e., overflow gets correctly set to 'auto', but then `overflow_x` and `overflow_y` being nothing erases the overflow value.

CC @SylvainCorlay. I think we should remove the `overflow` as a synced attribute, but for backwards compatibility we can make it a non-synced property that automatically sets `overflow_x` and `overflow_y`. Thoughts?

**Comment by harmon** [comment]
Yes! This caught me, and forced me to learn the CSSStyleDeclaration API (hehe: https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleDeclaration/removeProperty)

I was confused why overflow wasn't getting set, but when I defined all three, it worked! (As long as they were the same). Seems if any are null, the CSSStyleDeclaration.removeProperty() method will clear they all out, or at least the overflow one:

```
> parent.el.style['overflow'] = 'visible'
"visible"
> parent.el.style['overflow']
"visible"
> parent.el.style['overflow-x']
"visible"
> parent.el.style.removeProperty('overflow-x')
"visible"
> parent.el.style['overflow']
""
```

I think your suggestion to only have x and y as synced attributes makes the most sense, and I'd be happy with the backwards compatibility you proposed.

**Comment by SylvainCorlay** [comment]
I think that we kept `overlow-x` and `overflow-y` as traits because `overflow` is not really a shorthand property for `overflow-x` and `overflow-y` like `border` for `border-top`, `border-left` etc.

**Comment by jasongrout** [comment]
@SylvainCorlay - I'm not sure what you mean. The very first paragraph on MDN (https://developer.mozilla.org/en-US/docs/Web/CSS/overflow) says:

> The overflow CSS property specifies what to do when an element's content is too large to fit in its block formatting context. It is a shorthand for the overflow-x and overflow-y properties.

**Comment by SylvainCorlay** [comment]
It may be a browser support issue but you could not specify two keywords with overflow to give different values to overflow-x and overflow-y so we could not apply the policy to only expose shorthand properties that we had with others.

**Comment by jasongrout** [comment]
Indeed, https://developer.mozilla.org/en-US/docs/Web/CSS/overflow#Browser_compatibility indicates it only has support in some soon-to-be-released browsers
